### PR TITLE
OBJ-381 return what spec says will be returned

### PR DESCRIPTION
**Pull request by ICoxCH**

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/strikeoffobjections/model/eligibility/ObjectionEligibility.java [hunk 1]
```diff
@@ -5,13 +5,13 @@
...
-    private boolean isEligible;
+    private boolean eligible;
 
-    public ObjectionEligibility(boolean isEligible) {
-        this.isEligible = isEligible;
+    public ObjectionEligibility(boolean eligible) {
+        this.eligible = eligible;
     }
 
     public boolean isEligible() {
```
Comment (ICoxCH) [inline src/main/java/uk/gov/companieshouse/api/strikeoffobjections/model/eligibility/ObjectionEligibility.java:14]: Jackson assumes this corresponds to a field called eligible, as there wasn't one it was adding an "eligible" field to the returned json.
Changing the field to eligible removes this issue

File: src/main/java/uk/gov/companieshouse/api/strikeoffobjections/model/response/ObjectionResponseDTO.java [hunk 2]
```diff
@@ -30,6 +33,10 @@
     @JsonProperty("status")
     private ObjectionStatus status;
 
+    @JsonProperty("links")
+    @JsonUnwrapped
+    private Links links;
```
Comment (ICoxCH) [inline src/main/java/uk/gov/companieshouse/api/strikeoffobjections/model/response/ObjectionResponseDTO.java:38]: The spec mentions Objection has a links field, as the responseDTO is created using MapStruct, this was the best I could think of without more change, any suggestions welcome

File: src/main/java/uk/gov/companieshouse/api/strikeoffobjections/model/eligibility/ObjectionEligibility.java (hunk 1 above)
Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/api/strikeoffobjections/model/eligibility/ObjectionEligibility.java:14]: Just interested, how come this worked before? Didn't it mean we'd send a field back to the client that wasn't recognised or is there a change to do in 'web' too?

File: src/main/java/uk/gov/companieshouse/api/strikeoffobjections/model/response/ObjectionResponseDTO.java [hunk 3]
```diff
@@ -67,6 +75,7 @@ public String getCompanyNumber() {
 
     public void setCompanyNumber(String companyNumber) {
         this.companyNumber = companyNumber;
+        this.setLinks();
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/api/strikeoffobjections/model/response/ObjectionResponseDTO.java:78]: Doing this just doesn't look right. I'll have a look at code to understand the issue and see if I can see an alternative solution.

File: src/main/java/uk/gov/companieshouse/api/strikeoffobjections/model/eligibility/ObjectionEligibility.java (hunk 1 above)
Comment (ICoxCH) [reply src/main/java/uk/gov/companieshouse/api/strikeoffobjections/model/eligibility/ObjectionEligibility.java:14]: Jackson was sending two fields.
One it got from the JSON property annotation "is_eligible"
And another it got from inferring there should be a field called "eligible" from this getters name.

So the json response looked like
{
    is_eligible: true,
    eligible: true
}

I think the web was just ignoring the other field

File: src/main/java/uk/gov/companieshouse/api/strikeoffobjections/model/response/ObjectionResponseDTO.java (hunk 3 above)
Comment (ICoxCH) [reply src/main/java/uk/gov/companieshouse/api/strikeoffobjections/model/response/ObjectionResponseDTO.java:78]: Ideally we'd be able to add a little private method to the objection mapper "generateDTOLinks"
Then call objectionResponseDTO.setLinks(generateDTOLinks(ObjectionEntity); as part of the mapping.

Bu that's all abstracted away into mapstruct, and not sure if there's a way we could do that

File: src/main/java/uk/gov/companieshouse/api/strikeoffobjections/model/eligibility/ObjectionEligibility.java (hunk 1 above)
Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/api/strikeoffobjections/model/eligibility/ObjectionEligibility.java:14]: Ah right 👍 .

File: src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/ObjectionService.java [hunk 4]
```diff
@@ -283,11 +285,12 @@ public Attachment getAttachment(
         );
     }
 
-    private Links createLinks(String attachmentsUri, String attachmentId) {
-        String linkToSelf = attachmentsUri + "/" + attachmentId;
+    private Links createLinks(String selfLink, boolean download) {
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/ObjectionService.java:288]: Minor but could rename second var to 'createDownloadLink' which would improve readability in the 'if' statement lower down.

**Comment by mwestacott** [comment]
Looking good. One minor spot, plus you have conflicts to deal with.

**Review by mwestacott — APPROVED** [review]

**Review by markpit — APPROVED** [review]

**Pull request merged**